File: src/ros_basics_tutorials/src/iot_sensor/iot_sensor_sub.py
#!/usr/bin/env python2.7
import rospy
from ros_basics_tutorials.msg import IoT_Sensor

def iot_sensor_callback(iot_sensor_message):
    rospy.loginfo("new IoT data received: (%d, %s, %.2f ,%.2f,%.2f)", 
        iot_sensor_message.id,iot_sensor_message.name,
        iot_sensor_message.temperature,iot_sensor_message.humidity,iot_sensor_message.altitude)
    
rospy.init_node('iot_sensor_subscriber_node', anonymous=True)

rospy.Subscriber("iot_sensor_topic", IoT_Sensor, iot_sensor_callback)

# spin() simply keeps python from exiting until this node is stopped
rospy.spin()

File: src/ros_basics_tutorials/src/iot_sensor/IoT_SensorSUB.cpp
#include "ros/ros.h"
#include "ros_basics_tutorials/IoT_Sensor.h"

// Topic messages callback
void chatterCallback(const ros_basics_tutorials::IoT_Sensor::ConstPtr& msg)
{
    ROS_INFO("Subscriber : %d , %s, %.2f, %.2f, %.2f\n", msg->id,msg->name.c_str(),msg->temperature,msg->humidity,msg->altitude);
}

int main(int argc, char **argv)
{
    // Initiate a new ROS node named "listener"
	ros::init(argc, argv, "iot_sensor_sub_cpp_node");
	//create a node handle: it is reference assigned to a new node
	ros::NodeHandle node;


    // Subscribe to a given topic, in this case "chatter".
	//chatterCallback: is the name of the callback function that will be executed each time a message is received.
    ros::Subscriber sub = node.subscribe("chatter", 1000, chatterCallback);

    // Enter a loop, pumping callbacks
    ros::spin();

    return 0;
}
